compute/cker/include/cker/operation/BroadcastTo.h

```diff
@@ -0,0 +1,174 @@
+    if (can_use_32bit)
+    {
+      DoBCast32Bit<NDIMS>(device, output_tensor.template shaped<T, NDIMS>(bcast.result_shape()),
+                          input_tensor.template shaped<T, NDIMS>(bcast.x_reshape()),
+                          helper::BCast::ToIndexArrayType<int, NDIMS>(bcast.x_bcast()));
+    }
+    else
+    {
+      DoBCast<NDIMS>(device, output_tensor.template shaped<T, NDIMS>(bcast.result_shape()),
+                     input_tensor.template shaped<T, NDIMS>(bcast.x_reshape()),
+                     helper::BCast::ToIndexArrayType<Eigen::DenseIndex, NDIMS>(bcast.x_bcast()));
+    }
+  }
+
+  // PRECONDITION: rank(input_shape) > 0 &&
+  //               rank(input_shape) <= rank(output_shape)  &&
+  //               output_shape.num_elements() > 0.
+  void operator()(const Device &device, Tensor &output_tensor, const Shape &output_shape,
+                  const Tensor &input_tensor, const Shape &input_shape,
+                  const helper::BCast &bcast) const
+  {
+    const int ndims = bcast.y_reshape().size();
+    switch (ndims)
+    {
+      case 1:
+        ReshapeAndBCast<1>(device, output_tensor, input_tensor, bcast);
+        break;
+      case 2:
+        ReshapeAndBCast<2>(device, output_tensor, input_tensor, bcast);
+        break;
+      case 3:
+        ReshapeAndBCast<3>(device, output_tensor, input_tensor, bcast);
+        break;
+      case 4:
+        ReshapeAndBCast<4>(device, output_tensor, input_tensor, bcast);
+        break;
+      case 5:
+        ReshapeAndBCast<5>(device, output_tensor, input_tensor, bcast);
+        break;
+      default:
+        // NOTE : UNUSED leaves for maintenance purposes.
+        UNUSED(output_shape);
+        UNUSED(input_shape);
+        break;
+    }
+  }
+};
+} // functor
+
+template <typename T>
+inline void BroadcastTo(const Shape &input_shape, T *input_data, const Shape &output_shape,
+                        T *output_data)
+{
+  const int input_flatsize = input_shape.FlatSize();
+
+  if (input_shape == output_shape)
+  {
+    memcpy(output_data, input_data, input_flatsize * sizeof(T));
+    return;
+  }
+
+  assert(input_shape.FlatSize() <= output_shape.FlatSize());
+
+  Tensor output_tensor;
+  Tensor input_tensor;
+
+  input_tensor.shape.ReplaceWith(input_shape.DimensionsCount(), input_shape.DimsData());
+  input_tensor.buffer = input_data;
+
+  output_tensor.shape.ReplaceWith(output_shape.DimensionsCount(), output_shape.DimsData());
+  output_tensor.buffer = output_data;
+
+  const Eigen::ThreadPoolDevice &device = *eigen_support::GetThreadPoolDevice();
+
+  // Handle broadcast from Scalar.
+  if (input_flatsize == 0)
+  {
+    functor::FillFunctor<Eigen::ThreadPoolDevice, T>()(device, output_tensor.flat<T>(),
+                                                       input_tensor.scalar<T>());
+  }
+
+  helper::BCast bcast(helper::BCast::FromShape(input_shape), helper::BCast::FromShape(output_shape),
+                      /*fewer_dims_optimization=*/true);
+
```

Comment:
What happens when input and ouput are incompatible?
Don't we need to throw some exceptions?

Reply:
We can get some value which can know the calculation result via BCast's member method. 
So, I have added some assertion logic. 

Thanks.